hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenCreateValidator.java

```diff
@@ -183,12 +184,14 @@ private void validateSupplyType(final TokenSupplyType supplyType, final long max
     public void validateAssociation(
             @NonNull final EntitiesConfig entitiesConfig,
             @NonNull final TokensConfig tokensConfig,
+            @NonNull final ContractsConfig contractsConfig,
             @NonNull final Account account,
             @NonNull final Token token,
             @NonNull final WritableTokenRelationStore tokenRelStore) {
         validateFalse(
-                entitiesConfig.limitTokenAssociations()
-                        && account.numberAssociations() + 1 > tokensConfig.maxPerAccount(),
+                (entitiesConfig.limitTokenAssociations()
+                                && account.numberAssociations() + 1 > tokensConfig.maxPerAccount())
+                        || !contractsConfig.unlimitedAutoAssociations(),
```

Comment:
Do we need the unlimitedAutoAssociations, here? Aren't we overriding the limitTokenAssociations, one?